impactcommon/exogenous_economy/gdppc.py

```diff
@@ -1,55 +1,213 @@
     """
-    GDPpcProvider returns timeseries of GDPpc for a given IAM and SSP.
+    Read files on disk to create a BestGDPpcProvider instance
 
-    Usage example:
-      provider = GDPpcProvider('low', 'SSP3') # initialize the provider
-      gdppcs = provider.get_timeseries('ZWE.2.2') # get the series for any hierid or ISO
-      provider.get_startyear() # the first year
-    
-    Testing it:
-      1. Clone the impact-common repository to a folder.
-      2. Make a server.yml file in the same directory with the contents "shareddir: <path to GCP data dir.>"
-      3. cd into the impact-common directory and run:
-           python -m impactcommon.exogenous_economy.gdppc
+    Parameters
+    ----------
+    iam : str
+    ssp : str
+    growth_path_or_buffer
+        CSV file containing GDPpc growth projection at 5-year intervals. Must have columns giving "year",
+        IAM model ("model"), SSP scenario ("scenario"), the ISO entity ("iso"),
+        and the actual GDPpc growth values value ("growth").
+    baseline_path_or_buffer
+        CSV file containing GDPpc baseline values. Must have columns giving "year",
+        IAM model ("model"), SSP scenario ("scenario"), the ISO entity ("iso"),
+        and corresponding the actual GDPpc values ("value").
+    nightlights_path_or_buffer
+        CSV file of nightlight-based ratios for regions.
+    startyear : int, optional
+        Year to draw baseline value from. Must be in 'baseline_path_or_buffer' file's "year" column.
+    stopyear : int, optional
+        Must be within 5 years of the largest "year"  in 'growth_path_or_buffer'.
+    use_sharedpath : bool, optional
+        Interpret paths without leading "/" as "shareddir" paths?
+
+    Returns
+    -------
+    out : BestGDPpcProvider
+
+    Examples
+    --------
+    >>> provider = read_bestgdppcprovider(
+    ...     'low', 'SSP3',
+    ...     growth_path_or_buffer='inputdata/gdppc/growth.csv',
+    ...     baseline_path_or_buffer='inputdata/gdppc/baseline.csv',
+    ...     nightlights_path_or_buffer='inputdata/nightlights_normalized.csv'
+    ... )
+    >>> gdppcs = provider.get_timeseries('ZWE.2.2')  # Get the series for any hierid or ISO
+
+    See Also
+    --------
+    BestGDPpcProvider : Provider of GDP per capita (GDPpc) timeseries, selecting "best" data source
     """
-    
-    def __init__(self, iam, ssp):
-        """iam and ssp should be as described in the files (e.g., iam = 'low', ssp = 'SSP3')"""
+    if use_sharedpath:
+        baseline_path_or_buffer = files.sharedpath(baseline_path_or_buffer)
+        growth_path_or_buffer = files.sharedpath(growth_path_or_buffer)
+        nightlights_path_or_buffer = files.sharedpath(nightlights_path_or_buffer)
 
-        super(GDPpcProvider, self).__init__(iam, ssp, baseline_year)
+    df = metacsv.read_csv(baseline_path_or_buffer)
+    df_growth = metacsv.read_csv(files.sharedpath(growth_path_or_buffer))
```

Comment:
Do you mean for these calls to `sharedpath` to still be here?

Reply:
Ah. That's a bug.

Thank you! Fixed with `842f05ba`